tf2_ros_py/tf2_ros/buffer_client.py

```diff
@@ -44,11 +50,22 @@
 
 from tf2_msgs.action import LookupTransform
 
+# Used for documentation purposes only
+LookupTransformGoal = TypeVar('LookupTransformGoal')
+LookupTransformResult = TypeVar('LookupTransformResult')
```

Comment:
Could you explain a little more about why this is needed?

Reply:
The alternatives to get the type hints used with `__process_goal()` and `__process_result()` were:

1. Using `LookupTransform_Goal` and `LookupTransform_Result` by the below imports.

```python
from tf2_msgs.action._lookup_transform import LookupTransform_Goal
from tf2_msgs.action._lookup_transform import LookupTransform_Result
```
2. Using something more general like `ActionTypeGoal` and `ActionTypeResult`.

I decided against 1. as didn't feel appropriate to access the private `_lookup_transform` to get `LookupTransform_Goal` and against 2. for lack of information.

I am happy to change if there is something more appropriate.

The cpp version uses `LookupTransformAction::Goal` as you already know.

Reply:
Oh, I see.  Thanks for the explanation.  Looks good to me then.